### PR TITLE
PIL-835: restored manual sw activation paid by users

### PR DESCRIPTION
**Pull request by poocart**
https://linear.app/pillarproject/issue/PIL-835/restoration-of-user-financed-smart-wallet-deployment

### CODE REVIEW

File: src/services/smartWallet.js [hunk 1]
```diff
@@ -537,6 +539,13 @@ class SmartWallet {
     return formatEstimated(estimated);
   }
 
+  estimateAccountDeployment() {
+    return this.getSdk().estimateAccountDeployment().catch((err) => {
```
Comment (jsidorenko) [inline src/services/smartWallet.js:543]: If you catch the error here, then you won't be able to catch it again on the  top level

File: src/components/SWActivationModal/SWActivationModal.js [hunk 2]
```diff
@@ -18,99 +18,181 @@
...
+  const submitButtonDisabled = !isEnoughETH || !isOnline;
+  const submitButtonVisible = true;
+
+  let submitButtonTitle = t('button.activate');
+
+  if (gettingDeploymentEstimate) {
+    submitButtonTitle = t('label.gettingFee');
+  } else if (!isEnoughETH) {
+    submitButtonTitle = t('label.notEnoughToken', { token: ETH });
+  } if (!isOnline) {
```
Comment (IAmKio) [inline src/components/SWActivationModal/SWActivationModal.js:120]: Is there supposed to be a new line here?

File: src/services/smartWallet.js (hunk 1 above)
Comment (poocart) [reply src/services/smartWallet.js]: Was moving things around and left, fixed now, thanks 👍

**Review by IAmKio — APPROVED** [review]
One small comment re: code styling (it might also be my eyes) but other than that - LGTM!

File: src/components/SWActivationModal/SWActivationModal.js (hunk 2 above)
Comment (poocart) [reply src/components/SWActivationModal/SWActivationModal.js:120]: totally missed this, thanks

**Review by graszka22 — APPROVED** [review]
LGTM 👍

File: src/services/smartWallet.js [hunk 3]
```diff
@@ -256,8 +256,10 @@ class SmartWallet {
       .catch(e => this.reportError('Unable to sync smart wallets', { e }));
   }
 
-  async deployAccount(): Promise<{ error?: string, deployTxHash?: string }> {
-    const deployEstimate = await this.getSdk().estimateAccountDeployment().catch(this.handleError);
+  async deployAccount(
+    estimate?: sdkInterfaces.IEstimatedAccountDeployment,
+  ): Promise<{ error?: string, deployTxHash?: string }> {
+    const deployEstimate = estimate || await this.getSdk().estimateAccountDeployment().catch(this.handleError);
```
Comment (jsidorenko) [inline src/services/smartWallet.js:262]:
```suggestion
    const deployEstimate = estimate || await this.estimateAccountDeployment().catch(this.handleError);
```

Comment (poocart) [reply src/services/smartWallet.js]: 👍

**Review by jsidorenko — APPROVED** [review]
👍

**Pull request merged**